web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -0,0 +1,341 @@
+import { Component, ElementRef, Input, OnChanges, SimpleChanges, ViewChild, AbstractType} from '@angular/core';
+import { formatDate } from '@angular/common'
+import * as d3 from 'd3';
+
+import { step189_2020 } from '../../../proto/step189_2020';
+
+/**
+ * BarChartData includes all data used by the bar chart.
+ */
+interface BarChartData {
+  pushID: string; // Push ID number
+  state: number;  // Tag of push end state
+  startTime: any; // Start time of push, in `YYYY-MM-DD HH:mm` format
```

Comment:
Eventually we need to get rid of all instances of "any" in order to pass our tests. I feel like it's a pretty safe bet to assume "startTime" is of type "Date"